File: Selenium/Activities/Activity_14.java
package Activities;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;

import java.util.List;

public class Activity_14 {
    public static void main(String[] args) {
        WebDriver driver = new ChromeDriver();
        driver.get("https://v1.training-support.net/selenium/tables");
        driver.manage().window().maximize();
        System.out.println("The pagetitle is: " + driver.getTitle());
        List<WebElement> row = driver.findElements(By.xpath("//table[@id='sortableTable']/tbody/tr"));
        List<WebElement> column = driver.findElements(By.xpath("//table[@id='sortableTable']/tbody/tr[1]/td"));
        System.out.println("The number of rows: " + row.size());
        System.out.println("The number of columns: " + column.size());
        System.out.println("The value of second row and scond cell is: " + driver.findElement(By.xpath("//table[@id='sortableTable']/tbody/tr[2]/td[2]")).getText());
        driver.findElement(By.xpath("//table[@id='sortableTable']/thead//th[1]")).click();
        System.out.println("The value od second row and scond cell after sort is: " + driver.findElement(By.xpath("//table[@id='sortableTable']/tbody/tr[2]/td[2]")).getText());
        List<WebElement> tfooter = driver.findElements(By.xpath("//table[@id='sortableTable']//tfoot//th"));
        System.out.println("Values of table footer");
        for(WebElement ele:tfooter){
            System.out.println(ele.getText());
        }
        driver.quit();
    }
}

File: Selenium/Activities/Activity_14.py
from selenium import webdriver
from selenium.webdriver.common.by import By

with webdriver.Chrome() as driver:
    driver.get("https://v1.training-support.net/selenium/tables")
    print("The page title is: ", driver.title)
    driver.maximize_window()
    rows = driver.find_elements(By.XPATH, "//table[@id='sortableTable']/tbody/tr")
    columns = driver.find_elements(By.XPATH, "//table[@id='sortableTable']/tbody/tr[1]/td")
    print("The number of row in second table is: ", len(rows))
    print("The number of column in second table is: ", len(columns))
    print("The value of second row and second cell is: ",
          driver.find_element(By.XPATH, "//table[@id='sortableTable']/tbody/tr[2]/td[2]").text)
    driver.find_element(By.XPATH, "//table[@id='sortableTable']/thead//th[1]").click()
    print("The value of second row and second cell after sort is: ",
          driver.find_element(By.XPATH, "//table[@id='sortableTable']/tbody/tr[2]/td[2]").text)
    tfooter = driver.find_elements(By.XPATH, "//table[@id='sortableTable']//tfoot//th")
    for foo in tfooter:
        print(foo.text)
    driver.quit()